calliope/backend/pyomo/constraints/energy_balance.py

```diff
@@ -106,15 +106,16 @@ def system_balance_constraint_rule(backend_model, loc_carrier, timestep):
 
     """
     prod, con, export = get_loc_tech_carriers(backend_model, loc_carrier)
-    if hasattr(backend_model, 'unmet_demand'):
+    if hasattr(backend_model, 'bigM'):
```

Comment:
Are we sure that bigM is only ever an attribute if ``ensure_feasibility`` is set? There should be a comment here at least to explain why this check for bigM happens, since it's not immediately obvious?

Reply:
I could check for `ensure feasibility` here actually, I'll change that

Reply:
Updated in df65948